### PR TITLE
refactor: rpc types

### LINKED ISSUE #382: refactor(rpc): migrate rpc types from starknet-rs to types-rs
### Is there an existing issue?

- [X] I have searched the existing issues

### Motivation

- custom types are currently used for all database-stored data due to the reliance on bincode and serde for serialization
- types-rs is already utilized for felt, ensuring consistency and optimized serialization, as it aligns with other dependencies that also use this felt type
- for the feeder, we still rely on custom types since starknet-rs does not handle all serialization cases
- however, all RPCs currently depend on starknet-rs

### Request

transition the starknet-rs RPC types to types-rs to achieve uniformity, streamline serialization, and reduce dependency complexity

for reference, see the [starknet-types-rpc crate](https://github.com/starknet-io/types-rs/tree/main/crates/starknet-types-rpc)

### Solution

_No response_

### Are you willing to help with this request?

No.

### PR DESCRIPTION
**Pull request by jbcaron**
## Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Refactoring (no functional changes, no API changes)

## What is the current behavior?

Resolves: #382 #187 

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

Yes DB

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -8615,14 +8611,26 @@ dependencies = [
...
+ "num-traits",
+ "serde",
+]
+
 [[package]]
 name = "starknet-types-rpc"
 version = "0.7.1"
-source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "2d717a1b5ff67fa9ec13b09381513a4824fdd73ec5440fcc116088be6744aac4"
+source = "git+https://github.com/jbcaron/types-rs.git?branch=fork#234b5f03f91d7f791003e61ae1cfcfc1f096a539"
```
Comment (Trantorian1) [inline Cargo.lock]: Maybe we should move this to `madara-alliance`?

File: Cargo.lock [hunk 2]
```diff
@@ -8643,7 +8651,7 @@ dependencies = [
  "serde_json",
  "sha3",
  "starknet-crypto 0.5.2",
- "starknet-types-core",
+ "starknet-types-core 0.1.7 (git+https://github.com/kasarlabs/types-rs.git?branch=feat-deserialize-v0.1.7)",
```
Comment (Trantorian1) [inline Cargo.lock:8526]: Same for this repo

File: crates/client/exec/src/trace.rs [hunk 3]
```diff
@@ -48,27 +50,34 @@ pub fn execution_result_to_tx_trace(
...
+        keccak_builtin_applications: computation_resources.keccak_builtin_applications,
+        memory_holes: computation_resources.memory_holes,
+        pedersen_builtin_applications: computation_resources.pedersen_builtin_applications,
+        poseidon_builtin_applications: computation_resources.poseidon_builtin_applications,
+        range_check_builtin_applications: computation_resources.range_check_builtin_applications,
+        segment_arena_builtin: computation_resources.segment_arena_builtin,
+        steps: computation_resources.steps,
+        data_availability: starknet_types_rpc::DataAvailability {
+            l1_gas: execution_info.transaction_receipt.da_gas.l1_gas,
+            l1_data_gas: execution_info.transaction_receipt.da_gas.l1_data_gas,
```
Comment (Trantorian1) [inline crates/client/exec/src/trace.rs:66]: Since we are working off a fork, consider implementing `From`

File: crates/client/exec/src/trace.rs [hunk 4]
```diff
@@ -78,14 +87,14 @@ pub fn execution_result_to_tx_trace(
...
+            starknet_types_rpc::TransactionTrace::Invoke(starknet_types_rpc::InvokeTransactionTrace {
                 validate_invocation,
                 execute_invocation: if let Some(e) = &execution_info.revert_error {
-                    starknet_core::types::ExecuteInvocation::Reverted(starknet_core::types::RevertedInvocation {
+                    starknet_types_rpc::ExecuteInvocation::Anon(starknet_types_rpc::RevertedInvocation {
                         revert_reason: e.clone(),
                     })
                 } else {
-                    starknet_core::types::ExecuteInvocation::Success(
+                    starknet_types_rpc::ExecuteInvocation::FunctionInvocation(
```
Comment (Trantorian1) [inline crates/client/exec/src/trace.rs:97]: The naming for `ExecuteInvocation` enum members seems strange, maybe we should change this in our fork?

File: crates/client/rpc/src/test_utils.rs [hunk 5]
```diff
@@ -90,83 +90,108 @@ pub fn make_sample_chain_for_block_getters(backend: &MadaraBackend) -> SampleCha
...
-                finality_status: TransactionFinalityStatus::AcceptedOnL1,
+            TxnReceipt::<Felt>::Invoke(InvokeTxnReceipt {
+                common_receipt_properties: CommonReceiptProperties {
+                    transaction_hash: Felt::from_hex_unchecked("0x8888888"),
+                    actual_fee: FeePayment { amount: Felt::from_hex_unchecked("0x9"), unit: PriceUnit::Wei },
+                    messages_sent: vec![],
+                    events: vec![],
+                    execution_resources: defaut_execution_resources(),
+                    finality_status: TxnFinalityStatus::L1,
+                    anon: Anonymous::Successful(SuccessfulCommonReceiptProperties {
```
Comment (Trantorian1) [inline crates/client/rpc/src/test_utils.rs]: What does `Anonymous` refer to in this context? Just curious as to the terminology used by `type-rs`

File: crates/client/rpc/src/test_utils.rs [hunk 6]
```diff
@@ -90,83 +90,108 @@ pub fn make_sample_chain_for_block_getters(backend: &MadaraBackend) -> SampleCha
...
+            TxnReceipt::<Felt>::Invoke(InvokeTxnReceipt {
+                common_receipt_properties: CommonReceiptProperties {
+                    transaction_hash: Felt::from_hex_unchecked("0xdd848484"),
+                    actual_fee: FeePayment { amount: Felt::from_hex_unchecked("0x94"), unit: PriceUnit::Wei },
+                    messages_sent: vec![],
+                    events: vec![],
+                    execution_resources: defaut_execution_resources(),
+                    finality_status: TxnFinalityStatus::L2,
+                    anon: Anonymous::Successful(SuccessfulCommonReceiptProperties {
+                        execution_status: "SUCCEDED".into(),
```
Comment (Trantorian1) [inline crates/client/rpc/src/test_utils.rs]: Is there another way to do this than with String literals?

File: crates/client/rpc/src/test_utils.rs [hunk 7]
```diff
@@ -355,6 +380,22 @@ pub fn make_sample_chain_for_block_getters(backend: &MadaraBackend) -> SampleCha
...
+        memory_holes: None,
+        pedersen_builtin_applications: None,
+        poseidon_builtin_applications: None,
+        range_check_builtin_applications: None,
+        segment_arena_builtin: None,
+        steps: 0,
+        data_availability: starknet_types_rpc::DataAvailability { l1_data_gas: 0, l1_gas: 0 },
+    }
+}
+
```
Comment (Trantorian1) [inline crates/client/rpc/src/test_utils.rs:394]: Maybe we impl `Default` in the fork?

File: crates/client/rpc/src/versions/admin/v0_1_0/api.rs [hunk 8]
```diff
@@ -11,6 +11,6 @@ pub trait MadaraWriteRpcApi {
     #[method(name = "addDeclareV0Transaction")]
     async fn add_declare_v0_transaction(
         &self,
-        declare_transaction_v0: BroadcastedDeclareTransactionV0,
-    ) -> RpcResult<DeclareTransactionResult>;
+        _declare_v0_transaction: BroadcastedDeclareTransactionV0,
```
Comment (Trantorian1) [inline crates/client/rpc/src/versions/admin/v0_1_0/api.rs]: This field is still in use!

```suggestion
        declare_v0_transaction: BroadcastedDeclareTransactionV0,
```

File: crates/client/rpc/src/versions/user/v0_7_1/api.rs [hunk 9]
```diff
@@ -20,28 +21,38 @@ use starknet_types_core::felt::Felt;
...
+#[versioned_rpc("V0_7_1", "madara")]
+pub trait MadaraWriteRpcApi {
+    /// Submit a new class v0 declaration transaction
+    #[method(name = "addDeclareV0Transaction", and_versions = ["V0_8_0"])]
+    async fn add_declare_v0_transaction(
+        &self,
+        declare_transaction_v0: BroadcastedDeclareTransactionV0,
+    ) -> RpcResult<ClassAndTxnHash<Felt>>;
+}
+
```
Comment (Trantorian1) [inline crates/client/rpc/src/versions/user/v0_7_1/api.rs]: Careful, this has been moved to the admin method endpoint.

File: crates/client/rpc/src/versions/user/v0_7_1/methods/read/estimate_fee.rs [hunk 10]
```diff
@@ -43,7 +44,7 @@ pub async fn estimate_fee(
         .collect::<Result<Vec<_>, _>>()
         .or_internal_server_error("Failed to convert BroadcastedTransaction to AccountTransaction")?;
 
-    let validate = !simulation_flags.contains(&SimulationFlagForEstimateFee::SkipValidate);
+    let validate = !simulation_flags.contains(&"SKIP_VALIDATE".to_string());
```
Comment (Trantorian1) [inline crates/client/rpc/src/versions/user/v0_7_1/methods/read/estimate_fee.rs]: Same as before, is there a way we can avoid using String literals for this?

File: crates/client/rpc/src/versions/user/v0_7_1/methods/read/get_block_with_receipts.rs [hunk 11]
```diff
@@ -10,56 +11,58 @@ use crate::Starknet;
...
-                starknet_version: block.header.protocol_version.to_string(),
+                pending_block_header: PendingBlockHeader {
+                    parent_hash: block.header.parent_block_hash,
+                    timestamp: block.header.block_timestamp,
+                    sequencer_address: block.header.sequencer_address,
+                    l1_gas_price: block.header.l1_gas_price.l1_gas_price(),
+                    l1_data_gas_price: block.header.l1_gas_price.l1_data_gas_price(),
+                    l1_da_mode: block.header.l1_da_mode.into(),
+                    starknet_version: block.header.protocol_version.to_string(),
+                },
```
Comment (Trantorian1) [inline crates/client/rpc/src/versions/user/v0_7_1/methods/read/get_block_with_receipts.rs:46]: This is great :)

File: crates/client/rpc/src/versions/user/v0_7_1/methods/read/get_transaction_by_hash.rs [hunk 12]
```diff
@@ -41,9 +41,11 @@ pub fn get_transaction_by_hash(starknet: &Starknet, transaction_hash: Felt) -> S
     let transaction = block
         .inner
         .transactions
-        .get(tx_index.0 as usize)
+        .into_iter()
+        .nth(tx_index.0 as usize)
```
Comment (Trantorian1) [inline crates/client/rpc/src/versions/user/v0_7_1/methods/read/get_transaction_by_hash.rs:45]: Why `nth` instead of `get` here? Just curious.

File: crates/primitives/block/src/header.rs [hunk 13]
```diff
@@ -126,22 +127,23 @@ pub enum L1DataAvailabilityMode {
...
-}
+
+// impl From<starknet_core::types::L1DataAvailabilityMode> for L1DataAvailabilityMode {
+//     fn from(value: starknet_core::types::L1DataAvailabilityMode) -> Self {
+//         match value {
+//             starknet_core::types::L1DataAvailabilityMode::Calldata => Self::Calldata,
+//             starknet_core::types::L1DataAvailabilityMode::Blob => Self::Blob,
+//         }
+//     }
+// }
```
Comment (Trantorian1) [inline crates/primitives/block/src/header.rs]:
```suggestion
```

File: crates/primitives/class/src/class_hash.rs [hunk 14]
```diff
@@ -63,8 +63,9 @@ fn compute_hash_entries_point(entry_points: &[SierraEntryPoint]) -> Felt {
 
 impl CompressedLegacyContractClass {
     pub fn compute_class_hash(&self) -> Result<Felt, ComputeClassHashError> {
-        let legacy_contract_class = parse_compressed_legacy_class(self.clone().into())?;
-        legacy_contract_class.class_hash().map_err(ComputeClassHashError::from)
+        unimplemented!()
+        // let legacy_contract_class = parse_compressed_legacy_class(self.clone().into())?;
+        // legacy_contract_class.class_hash().map_err(ComputeClassHashError::from)
```
Comment (Trantorian1) [inline crates/primitives/class/src/class_hash.rs]: Why was this removed?

File: crates/primitives/class/src/into_starknet_types.rs [hunk 15]
```diff
@@ -0,0 +1,479 @@
...
+impl From<starknet_types_rpc::EventAbiType> for LegacyEventAbiType {
+    fn from(_: starknet_types_rpc::EventAbiType) -> Self {
+        LegacyEventAbiType::Event
+    }
+}
+
+impl From<LegacyEventAbiType> for starknet_types_rpc::EventAbiType {
+    fn from(legacy_event_abi_type: LegacyEventAbiType) -> Self {
+        match legacy_event_abi_type {
+            LegacyEventAbiType::Event => "event".to_string(),
```
Comment (Trantorian1) [inline crates/primitives/class/src/into_starknet_types.rs:392]: ?

File: crates/primitives/class/src/into_starknet_types.rs [hunk 16]
```diff
@@ -0,0 +1,479 @@
...
+impl From<starknet_types_rpc::StructAbiType> for LegacyStructAbiType {
+    fn from(_: starknet_types_rpc::StructAbiType) -> Self {
+        LegacyStructAbiType::Struct
+    }
+}
+
+impl From<LegacyStructAbiType> for starknet_types_rpc::StructAbiType {
+    fn from(legacy_struct_abi_type: LegacyStructAbiType) -> Self {
+        match legacy_struct_abi_type {
+            LegacyStructAbiType::Struct => "struct".to_string(),
```
Comment (Trantorian1) [inline crates/primitives/class/src/into_starknet_types.rs:406]: Same here

File: crates/primitives/class/src/into_starknet_types.rs [hunk 17]
```diff
@@ -0,0 +1,479 @@
...
+impl From<starknet_types_rpc::FunctionStateMutability> for FunctionStateMutability {
+    fn from(_: starknet_types_rpc::FunctionStateMutability) -> Self {
+        FunctionStateMutability::View
+    }
+}
+
+impl From<FunctionStateMutability> for starknet_types_rpc::FunctionStateMutability {
+    fn from(function_state_mutability: FunctionStateMutability) -> Self {
+        match function_state_mutability {
+            FunctionStateMutability::View => "view".to_string(),
```
Comment (Trantorian1) [inline crates/primitives/class/src/into_starknet_types.rs:420]: Same here

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 18]
```diff
@@ -0,0 +1,389 @@
...
+//             starknet_types_rpc::TxnReceipt::<Felt>::L1Handler(receipt) => TransactionReceipt::L1Handler(receipt.into()),
+//             starknet_types_rpc::TxnReceipt::<Felt>::Declare(receipt) => TransactionReceipt::Declare(receipt.into()),
+//             starknet_types_rpc::TxnReceipt::<Felt>::Deploy(receipt) => TransactionReceipt::Deploy(receipt.into()),
+//             starknet_types_rpc::TxnReceipt::<Felt>::DeployAccount(receipt) => {
+//                 TransactionReceipt::DeployAccount(receipt.into())
+//             }
+//         }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 19]
```diff
@@ -0,0 +1,389 @@
...
+//             transaction_hash: receipt.common_receipt_properties.transaction_hash,
+//             actual_fee: receipt.common_receipt_properties.actual_fee.into(),
+//             messages_sent: receipt.common_receipt_properties.messages_sent.into_iter().map(MsgToL1::from).collect(),
+//             events: receipt.common_receipt_properties.events.into_iter().map(Event::from).collect(),
+//             execution_resources: receipt.common_receipt_properties.execution_resources.into(),
+//             execution_result: receipt.common_receipt_properties.anon.into(),
+//         }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 20]
```diff
@@ -0,0 +1,389 @@
...
+//             transaction_hash: receipt.transaction_hash,
+//             actual_fee: receipt.actual_fee.into(),
+//             messages_sent: receipt.messages_sent.into_iter().map(MsgToL1::from).collect(),
+//             events: receipt.events.into_iter().map(Event::from).collect(),
+//             execution_resources: receipt.execution_resources.into(),
+//             execution_result: receipt.execution_result.into(),
+//         }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 21]
```diff
@@ -0,0 +1,389 @@
...
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 22]
```diff
@@ -0,0 +1,389 @@
...
+//             actual_fee: receipt.actual_fee.into(),
+//             messages_sent: receipt.messages_sent.into_iter().map(MsgToL1::from).collect(),
+//             events: receipt.events.into_iter().map(Event::from).collect(),
+//             execution_resources: receipt.execution_resources.into(),
+//             execution_result: receipt.execution_result.into(),
+//             contract_address: receipt.contract_address,
+//         }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 23]
```diff
@@ -0,0 +1,389 @@
...
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 24]
```diff
@@ -0,0 +1,389 @@
...
+        }
+    }
+}
+
+// impl From<starknet_types_rpc::FeePayment<Felt>> for FeePayment {
+//     fn from(payment: starknet_types_rpc::FeePayment<Felt>) -> Self {
+//         Self { amount: payment.amount, unit: payment.unit.into() }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 25]
```diff
@@ -0,0 +1,389 @@
...
+// impl From<starknet_types_rpc::PriceUnit> for PriceUnit {
+//     fn from(unit: starknet_types_rpc::PriceUnit) -> Self {
+//         match unit {
+//             starknet_types_rpc::PriceUnit::Fri => PriceUnit::Fri,
+//             starknet_types_rpc::PriceUnit::Wei => PriceUnit::Wei,
+//         }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 26]
```diff
@@ -0,0 +1,389 @@
...
+        }
+    }
+}
+
+// impl From<starknet_types_rpc::MsgToL1<Felt>> for MsgToL1 {
+//     fn from(msg: starknet_types_rpc::MsgToL1<Felt>) -> Self {
+//         Self { from_address: msg.from_address, to_address: msg.to_address, payload: msg.payload }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 27]
```diff
@@ -0,0 +1,389 @@
...
+        Self { from_address: msg.from_address, to_address: msg.to_address, payload: msg.payload }
+    }
+}
+
+// impl From<starknet_types_rpc::Event<Felt>> for Event {
+//     fn from(event: starknet_types_rpc::Event<Felt>) -> Self {
+//         Self { from_address: event.from_address, keys: event.keys, data: event.data }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 28]
```diff
@@ -0,0 +1,389 @@
...
+//             ecdsa_builtin_applications: computation_resources.ecdsa_builtin_applications,
+//             bitwise_builtin_applications: computation_resources.bitwise_builtin_applications,
+//             keccak_builtin_applications: computation_resources.keccak_builtin_applications,
+//             segment_arena_builtin: computation_resources.segment_arena_builtin,
+//             data_availability: resources.data_availability.into(),
+//             total_gas_consumed: Default::default(),
+//         }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 29]
```diff
@@ -0,0 +1,389 @@
...
+        }
+    }
+}
+
+// impl From<starknet_types_rpc::DataAvailability> for L1Gas {
+//     fn from(resources: starknet_types_rpc::DataAvailability) -> Self {
+//         Self { l1_gas: resources.l1_gas.into(), l1_data_gas: resources.l1_data_gas.into() }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 30]
```diff
@@ -0,0 +1,389 @@
...
+//         match result {
+//             starknet_types_rpc::Anonymous::Successful(_) => ExecutionResult::Succeeded,
+//             starknet_types_rpc::Anonymous::Reverted(starknet_types_rpc::RevertedCommonReceiptProperties {
+//                 revert_reason,
+//                 ..
+//             }) => ExecutionResult::Reverted { reason: revert_reason },
+//         }
+//     }
+// }
+
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]:
```suggestion
```

File: crates/primitives/receipt/src/into_starknet_types.rs [hunk 31]
```diff
@@ -0,0 +1,389 @@
...
+//         let core_receipt = receipt.clone().to_starknet_types(starknet_types_rpc::TxnFinalityStatus::L1);
+//         let receipt_back: TransactionReceipt = core_receipt.into();
+//         assert_eq!(receipt, receipt_back);
+
+//         let receipt: TransactionReceipt = dummy_deploy_account_receipt().into();
+//         let core_receipt = receipt.clone().to_starknet_types(starknet_types_rpc::TxnFinalityStatus::L1);
+//         let receipt_back: TransactionReceipt = core_receipt.into();
+//         assert_eq!(receipt, receipt_back);
+//     }
+// }
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/into_starknet_types.rs]: Why remove these tests?

File: crates/primitives/transactions/src/broadcasted_to_blockifier.rs [hunk 32]
```diff
@@ -76,88 +79,89 @@ pub fn broadcasted_declare_v0_to_blockifier(
...
-                        info: class_info,
-                        compiled: Arc::new(compiled),
-                    })),
-                )
             }
-        },
-        _ => (None, None, None),
-    };
+            _ => (None, None, None),
+        };
```
Comment (Trantorian1) [inline crates/primitives/transactions/src/broadcasted_to_blockifier.rs]: Consider refactoring this into sub-functions to make it more readable, I'm havin a hard time sifting through this on review :face_with_spiral_eyes:

**Review by Trantorian1 — COMMENTED** [review]
Awesome work, seems like this will be much more pleasant to develop with in the future :)

File: crates/client/exec/src/trace.rs (hunk 4 above)
Comment (jbcaron) [reply crates/client/exec/src/trace.rs:97]: fixed

File: crates/client/rpc/src/test_utils.rs (hunk 5 above)
Comment (jbcaron) [reply crates/client/rpc/src/test_utils.rs]: fixed

File: crates/client/rpc/src/test_utils.rs (hunk 6 above)
Comment (jbcaron) [reply crates/client/rpc/src/test_utils.rs]: fixed

```rust
// The execution status of the transaction
#[derive(Eq, Hash, PartialEq, Serialize, Deserialize, Clone, Debug)]
#[serde(tag = "execution_status", content = "revert_reason")]
pub enum ExecutionStatus {
    #[serde(rename = "SUCCEEDED")]
    Successful,
    #[serde(rename = "REVERTED")]
    Reverted(String),
}
```

File: crates/client/rpc/src/versions/admin/v0_1_0/api.rs (hunk 8 above)
Comment (jbcaron) [reply crates/client/rpc/src/versions/admin/v0_1_0/api.rs]: fixed

File: crates/client/rpc/src/versions/user/v0_7_1/api.rs (hunk 9 above)
Comment (jbcaron) [reply crates/client/rpc/src/versions/user/v0_7_1/api.rs]: removed

File: crates/client/rpc/src/versions/user/v0_7_1/methods/read/get_transaction_by_hash.rs (hunk 12 above)
Comment (jbcaron) [reply crates/client/rpc/src/versions/user/v0_7_1/methods/read/get_transaction_by_hash.rs:45]: so as not to have a reference

File: crates/client/rpc/src/versions/user/v0_7_1/methods/read/estimate_fee.rs (hunk 10 above)
Comment (jbcaron) [reply crates/client/rpc/src/versions/user/v0_7_1/methods/read/estimate_fee.rs]: fixed

File: crates/primitives/receipt/src/to_starknet_types.rs [hunk 33]
```diff
@@ -0,0 +1,203 @@
...
+            poseidon_builtin_applications: nullify_zero(resources.poseidon_builtin_applications),
+            range_check_builtin_applications: nullify_zero(resources.range_check_builtin_applications),
+            segment_arena_builtin: nullify_zero(resources.segment_arena_builtin),
+            steps: resources.steps,
+            data_availability: resources.data_availability.into(),
+        }
+    }
+}
+
+fn nullify_zero(u: u64) -> Option<u64> {
```
Comment (Trantorian1) [inline crates/primitives/receipt/src/to_starknet_types.rs:183]: Consider making this a helper function inside of `impl From<ExecutionResources> for starknet_types_rpc::ExecutionResources` (nitpick)

**Review by Trantorian1 — APPROVED** [review]
Seems great!

**Review by antiyro — APPROVED** [review]
tested locally works fine on e2e test read calls
